docs/reference-guides/data/data-core-editor.md

```diff
@@ -300,7 +300,7 @@ _Parameters_
 
 _Returns_
 
--   `string?`: Preview Link.
```

Comment:
Is this notation not allowed anymore?

Reply:
@youknowriad Based on my digging, there are two separate things at play here:

1. Nullable returns (`?<type>`) are causing an error if the return value is not literally the type or `null`. Which is the case when the function returns just `return;` (`undefined`)

This here is invalid:
```js
/**
 * only return string if it isn't equal to 'test'
 * @param {string} 
 * @return {?string}
 */
export const foo = ( input ) => {
 
    if ( 'test' === input ) {
        return;
    }

    return input;
};
```

This here is valid:
```js
/**
 * only return string if it isn't equal to 'test'
 * @param {string} 
 * @return {?string}
 */
export const foo = ( input ) => {
 
    if ( 'test' === input ) {
        return null;
    }

    return input;
};
```

3. Optional returns don't seem to have ever really been a thing but JSDoc didn't catch it correctly before. 

I cannot find any reference to the `type?` syntax anywhere in JSDoc.

Reply:
Thanks for the explanations, that makes sense.

Reply:
There's probably some tweaks that we could make to improve these edge cases for these functions (like changing the return values to null or empty strings...) but I think for the purpose of this PR. Your changes are good.

Reply:
Currently to keep the scope of this PR as small as it could possibly be I limited my changes to only the doc comments in the files where `npm run lint:js` threw an error after updating to the latest version of the package.